### DISCUSSION TITLE
The Best Way To Handle CPU-Controlled Branch In VGMSeq?

### DISCUSSION THREAD

**Discussion by loveemu** [vgmtrans/vgmtrans#272] [General]
A song sequence sometimes may have a special jump event. The jump condition is determined from external input (CPU variable) and the song syncs to the situation of the game, as a result. (for example, an intro part repeats until a certain conversation has begun)

It is hard to parse such songs without external input. I think a feature that can write to such a variable from an external script (Lua?) may be useful.

Any good ideas to handle such cases?

**Comment by KungFuFurby** [comment]
You'll have to handle these scenarios on a case-by-case basis. Technically depending on what it is you could create a switch of some kind that handles these kind of cases if they're merely on/off switches (or even just a numerical comparison against a single variable, but there are other cases that are more complicated.

If I run into a CPU-controlled branch (or find a setting that the SNES can set to control such a branch), then I can document the SNES-side command(s) that control the branch.

**Comment by Meerkov** [comment]
To be honest, if you're going to go through the effort of simulating the CPU's control branches in LUA on a game by game basis, there's likely a real suggestion to prefer *starting* with a full working emulator, and extracting the music from there, rather than starting with something narrowly focused on music and trying to build an emulator from scratch.
